### DISCUSSION TITLE
Supabase Edge functions + Internal package in Turborepo

### DISCUSSION THREAD

**Discussion by vext-consulting** [supabase/supabase#26923] [Questions]
Hi there 👋 First of all, this is the project structure within my Turborepo:

## Project Structure

```bash
apps/
├── docs/
├── edge-functions/
│   ├── node_modules/
│   ├── supabase/
│   └── package.json
├── server/
└── web/
```

I am currently setting up edge functions locally. I have an internal package that is currently used by both my web app and my server app. Now, having added edge-functions, I want to use my internal package in the edge functions as well. However, I am encountering the following error:

`{"code":"BOOT_ERROR", "message":"Worker failed to boot (please check logs)"}`

I have explored node compatibility, especially the [resource from Supabase](https://supabase.com/blog/edge-functions-node-npm), but haven't had any luck. I also tried bundling the package locally and moving it from the /packages directory to apps/edge-functions, but I find this approach very cumbersome.

Any suggestions would be much appreciated!

**Comment by eybel** [comment]
I am having the same problem. I am tryin to run a function locally but I get that error message.
My code:


import { serve } from "https://deno.land/std@0.168.0/http/server.ts";
import { stripe } from "./_utils/stripe";

serve(async (req) => {
  try {
    const { amount } = await req.json();
    const paymentIntent = await stripe.paymentIntents.create({
      amount: 1099,
      currency: "usd",
    });

    const res = {
      paymentIntent: paymentIntent.client_secret,
      publishableKey: Deno.env.get("EXPO_PUBLIC_STIPE_PUBLISHABLE_KEY"),
    };

    return new Response(JSON.stringify(res), {
      headers: { "Content-Type": "application/json" },
    });

  } catch (error) {

    return new Response(JSON.stringify(error), {
      headers: { "Content-Type": "application/json" },
      status: 400,
    });
  }
});

```
/* To invoke locally:

  1. Run `supabase start` (see: https://supabase.com/docs/reference/cli/supabase-start)
  2. Make an HTTP request:

  curl -i --location --request POST 'http://127.0.0.1:543/functions/v1/payment-sheet' \
    --header 'Authorization: Bearer eLJKASDJKHASDJKHASKJDHASKJHDJKASHDJKASHDJKAHSJKDHAJKSD' \
    --header 'Content-Type: application/json' \
    --data '{"amount": 1150}'

*/

```

**Reply by vext-consulting** [reply]
I see. I would recommend you to explore the Supabase Discord for answers as well, as it seems like there are some similar issues.

**Reply by AbhirKarande** [reply]
Have you figured out a solution to this?
